plugins/modules/content_export_info.py

```diff
@@ -0,0 +1,165 @@
+    type: complete
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+- name: "Get a specific export history and register the result for the next task"
+  vars:
+    organization_name: "Export Org"
+  content_export_info:
+    id: 29
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+  register: result
+- name: "Write metadata.json to disk using data from the previous task"
+  vars:
+    metadata: "{{ result['task']['results'][0]['metadata'] }}"
+  ansible.builtin.copy:
+    content: "{{ metadata }}"
+    dest: ./metadata.json
+- name: "List all exports of a specific content view version"
+  content_export_info:
+    content_view_version_id: 379
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+- name: "List all exports marked for a specific destination server"
+  content_export_info:
+    destination_server: "airgapped.example.com"
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+- name: "List incremental exports of a specific content view version marked for a specific destination server"
+  content_export_info:
+    content_view_id: 1
+    destination_server: "airgapped.example.com"
+    type: incremental
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+- name: "List all exports of a specific content view marked for a specific destination server"
+  content_export_info:
+    content_view_id: 1
+    destination_server: "airgapped.example.com"
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloAnsibleModule, _flatten_entity
+
+
+class KatelloContentExportInfoModule(KatelloAnsibleModule):
+    pass
+
+
+def main():
+    module = KatelloContentExportInfoModule(
+        foreman_spec=dict(
+            id=dict(required=False, type='int'),
+            content_view_version_id=dict(required=False, type='int'),
+            content_view_id=dict(required=False, type='int'),
+            destination_server=dict(required=False, type='str'),
+            organization_id=dict(required=False, type='int'),
+            type=dict(required=False, type='str', choices=['complete', 'incremental']),
+            search=dict(required=False, type='str'),
+        ),
+    )
+
+    with module.api_connection():
+        module.auto_lookup_entities()
+
+        endpoint = 'content_exports'
+
+        payload = _flatten_entity(module.foreman_params, module.foreman_spec)
+        task = module.resource_action(endpoint, 'index', payload)
+
+        module.exit_json(task=task)
```

Comment:
We have a shortcut for `resource_action('index')`: `list_resource`, it also takes care of stripping the `results` "outer-array".

```suggestion
        resource = 'content_exports'

        payload = _flatten_entity(module.foreman_params, module.foreman_spec)
        info = {resource: module.list_resource(resource, params=payload)}

        module.exit_json(**info)
```

We could completely avoid this code, if we would base the module off `ForemanInfoAnsibleModule` and mark `name` as `invisible`. Yes!

```python
from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloInfoAnsibleModule


class KatelloContentExportInfo(KatelloInfoAnsibleModule):
    pass


def main():
    module = KatelloContentExportInfo(
        foreman_spec=dict(
            id=dict(required=False, type='int'),
            content_view_version=dict(type='entity', scope=['content_view'], required=False),
            content_view=dict(type='entity', scope=['organization'], required=False),
            destination_server=dict(required=False, type='str'),
            type=dict(required=False, type='str', choices=['complete', 'incremental']),
            search=dict(required=False, type='str'),
            name=dict(invisible=True),
        ),
    )

    with module.api_connection():
        module.run()


if __name__ == '__main__':
    main()
```

that works :)